src/student_success_tool/analysis/pdp/targets/shared.py

```diff
@@ -281,6 +283,55 @@ def _compute_intensity_num_terms(
     ]
 
 
+def get_nth_student_terms(
+    df: pd.DataFrame,
+    *,
+    student_id_cols: str | list[str] = "student_guid",
+    n: int = 1,
+    sort_cols: str | list[str] = "term_rank",
+    include_cols: t.Optional[list[str]] = None,
+    term_is_pre_cohort_col: str = "term_is_pre_cohort",
+    exclude_pre_cohort_terms: bool = True,
+) -> pd.DataFrame:
+    """
+    For each student, get the nth row in ``df`` (in ascending order of ``sort_cols`` ). If `exclude_pre_cohort_col` is true, then for each student, we want to get the nth row in ``df`` (in ascending order of ``sort_cols`` ) for which the term occurred *within* the student's cohort, i.e. not prior to their official start of enrollment, and a configurable subset of columns.
+    Ex. n = 0 gets the first term, and is equivalent to the functionality of get_first_student_terms(); n = 1 gets the second term, n = 2, gets the third term, so on and so forth.
+
+    Args:
+        df
+        student_id_cols
+        n
+        sort_cols
+        include_cols
+        term_is_pre_cohort_col
+        exclude_pre_cohort_terms
+    """
+    student_id_cols = utils.to_list(student_id_cols)
+    sort_cols = utils.to_list(sort_cols)
+    cols = (
+        df.columns.tolist()
+        if include_cols is None
+        else list(
+            utils.unique_elements_in_order(student_id_cols + sort_cols + include_cols)
+        )
+    )
+    if exclude_pre_cohort_terms == True:
```

Comment:
`is True` and `is False` (and `is None`) are idiomatic Python conditions -- don't ever use `== True`, unless a package specifically wants that construction in certain contexts (e.g. `sqlalchemy`, `numpy`).

```suggestion
    if exclude_pre_cohort_terms is True:
```